sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2184,7 +2184,9 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    * Create a Spark DataType.
    */
   private def visitSparkDataType(ctx: DataTypeContext): DataType = {
-    HiveStringType.replaceCharType(typedVisit(ctx))
+    HiveNullType.replaceNullType(
+      HiveStringType.replaceCharType(typedVisit(ctx))
+    )
```

Comment:
One line is enough, isn't it?
```scala
HiveNullType.replaceNullType(HiveStringType.replaceCharType(typedVisit(ctx)))
```

Reply:
Yes. Split to two lines is just for readability. Ok, I will coalesce to one line.